packages/components/src/dropdown-menu/dropdown-menu.tsx

```diff
@@ -207,6 +208,49 @@ export const CheckboxItem = forwardRef<
 
 CheckboxItem.displayName = DropdownMenu.CheckboxItem.displayName
 
+/**
+ * SwitchItem
+ */
+
+type SwitchItemProps = DropdownMenu.DropdownMenuCheckboxItemProps & {
+  icon?: IconElement
+  label: string
+  checked: boolean
+  onCheckedChange: (checked: boolean) => void
+  danger?: boolean
+}
+
+export const SwitchItem = forwardRef<
+  React.ElementRef<typeof DropdownMenu.CheckboxItem>,
+  SwitchItemProps
+>((props, ref) => {
+  const { label, icon, danger, ...itemProps } = props
+
+  const id = useId()
+
+  return (
+    <DropdownMenu.CheckboxItem
+      ref={ref}
+      className={itemStyles()}
+      {...itemProps}
+    >
+      {icon && (
+        <span className={iconStyles({ danger })}>{cloneElement(icon)}</span>
+      )}
+      <span className={labelStyles({ danger })}>{label}</span>
+      <Switch
+        id={id}
+        checked={itemProps.checked}
+        onCheckedChange={() => {
+          // handled by parent
+        }}
+      />
+    </DropdownMenu.CheckboxItem>
+  )
+})
+
+SwitchItem.displayName = DropdownMenu.CheckboxItem.displayName
```

Comment:
Is it ok that `SwitchItem` has same displayName as `CheckboxItem`?

Reply:
thx https://github.com/status-im/status-web/pull/582/commits/83b7f75ba90f4a9088a60a2fe955efa8793c05db